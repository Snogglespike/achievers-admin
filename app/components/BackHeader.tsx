import { Link } from "@remix-run/react";

import { ArrowLeft } from "iconoir-react";

interface Props {
  to?: string;
}

export function BackHeader({ to = "../" }: Props) {
  return (
    <>
      <div>
        <Link to={to} relative="path" className="btn-ghost btn mb-2 gap-2">
          <ArrowLeft className="w-6" />
          Back
        </Link>
      </div>

      <hr className="mb-4" />
    </>
  );
}
