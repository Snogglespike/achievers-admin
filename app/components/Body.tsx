import type { AzureUserWebAppWithRole } from "~/services";

import { Outlet } from "@remix-run/react";

import { Drawer } from "./Drawer";
import { Navbar } from "./Navbar";

interface Props {
  isAdmin: boolean;
  hasCompletedVolunteerAgreement?: boolean;
  version: string;
  currentUser: AzureUserWebAppWithRole;
}

export function Body({
  currentUser,
  hasCompletedVolunteerAgreement,
  isAdmin,
  version,
}: Props) {
  const showDrawer = isAdmin || hasCompletedVolunteerAgreement;

  return (
    <div className="drawer-mobile drawer">
      <input id="drawer" type="checkbox" className="drawer-toggle" />
      <div className="drawer-content flex flex-col">
        <Navbar currentUser={currentUser} />

        <main className="mt-16 flex h-full flex-col overflow-y-auto bg-white p-4">
          <Outlet />
        </main>
      </div>

      {showDrawer && <Drawer isAdmin={isAdmin} version={version} />}
    </div>
  );
}
