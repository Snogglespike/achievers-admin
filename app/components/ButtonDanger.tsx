import type { ButtonHTMLAttributes, DetailedHTMLProps } from "react";

export default function ButtonDanger({
  children,
  className = "",
  disabled,
  ...rest
}: DetailedHTMLProps<
  ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
>) {
  return (
    <button
      {...rest}
      disabled={disabled}
      className={
        disabled
          ? "flex items-center justify-center rounded border bg-red-200 py-2 px-4 text-white " +
            className
          : "flex items-center justify-center rounded border bg-red-500 py-2 px-4 text-white hover:bg-red-600 " +
            className
      }
    >
      {children}
    </button>
  );
}
