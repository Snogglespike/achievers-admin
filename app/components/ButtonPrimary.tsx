import type { ButtonHTMLAttributes, DetailedHTMLProps } from "react";

export default function ButtonPrimary({
  children,
  className = "",
  disabled,
  ...rest
}: DetailedHTMLProps<
  ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
>) {
  return (
    <button
      {...rest}
      disabled={disabled}
      className={
        disabled
          ? "flex items-center justify-center rounded bg-blue-200 py-2 px-4 text-white " +
            className
          : "flex items-center justify-center rounded bg-blue-500 py-2 px-4 text-white hover:bg-blue-600 " +
            className
      }
    >
      {children}
    </button>
  );
}
