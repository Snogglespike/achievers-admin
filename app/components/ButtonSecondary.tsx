import type { ButtonHTMLAttributes, DetailedHTMLProps } from "react";

export default function ButtonSecondary({
  children,
  className = "",
  disabled,
  ...rest
}: DetailedHTMLProps<
  ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
>) {
  return (
    <button
      {...rest}
      disabled={disabled}
      className={
        disabled
          ? "flex items-center justify-center rounded bg-slate-100 py-2 px-4 text-slate-400 " +
            className
          : "flex items-center justify-center rounded border border-slate-300 bg-slate-200 px-4 py-2 hover:bg-slate-300 " +
            className
      }
    >
      {children}
    </button>
  );
}
