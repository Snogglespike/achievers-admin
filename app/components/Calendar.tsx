import type { Dayjs } from "dayjs";
import dayjs from "dayjs";

import { ArrowRightIcon, ArrowLeftIcon } from "@heroicons/react/24/solid";

interface Props {
  selectedDate: Dayjs;
  previousMonth: () => void;
  nextMonth: () => void;
  generateWeeksOfTheMonth: Date[][];
  currentDay: Date;
}

export default function Calendar({
  selectedDate,
  previousMonth,
  nextMonth,
  generateWeeksOfTheMonth,
  currentDay,
}: Props) {
  return (
    <div className="hidden w-full max-w-sm md:block">
      <div className="rounded-t bg-white p-5 md:p-8">
        <div className="flex items-center justify-between px-4">
          <span
            tabIndex={0}
            className="text-base font-bold text-gray-800 focus:outline-none"
          >
            {selectedDate.clone().format("YYYY MMM")}
          </span>
          <div className="flex items-center">
            <button
              aria-label="calendar backward"
              className="text-gray-800 hover:text-gray-400 focus:text-gray-400"
              onClick={previousMonth}
            >
              <ArrowLeftIcon className="w-6 text-blue-600" />
            </button>
            <button
              aria-label="calendar forward"
              className="ml-3 text-gray-800 hover:text-gray-400 focus:text-gray-400"
              onClick={nextMonth}
            >
              <ArrowRightIcon className="w-6 text-blue-600" />
            </button>
          </div>
        </div>
        <div className="flex items-center justify-between overflow-x-auto pt-12">
          <table className="w-full">
            <thead>
              <tr>
                {generateWeeksOfTheMonth[0].map((day, index) => (
                  <th key={`week-day-${index}`}>
                    <div className="flex w-full justify-center">
                      <p className="text-center text-base font-medium text-gray-800">
                        {dayjs(day).format("dd")}
                      </p>
                    </div>
                  </th>
                ))}
              </tr>
            </thead>
            <tbody>
              {generateWeeksOfTheMonth.map((week, weekIndex) => (
                <tr key={`week-${weekIndex}`}>
                  {week.map((day, dayIndex) => (
                    <td className="pt-6" key={`day-${dayIndex}`}>
                      {dayjs(currentDay).isSame(day, "date") ? (
                        <div className="h-full w-full">
                          <div className="flex w-full cursor-pointer items-center justify-center rounded-full">
                            <span
                              role="link"
                              tabIndex={0}
                              className="flex  h-8 w-8 items-center justify-center rounded-full bg-indigo-700 text-base font-medium text-white hover:bg-indigo-500 focus:bg-indigo-500 focus:outline-none focus:ring-2 focus:ring-indigo-700 focus:ring-offset-2"
                            >
                              {day.getDate()}
                            </span>
                          </div>
                        </div>
                      ) : (
                        <div className="flex w-full cursor-pointer justify-center px-2 py-2">
                          <p
                            className={
                              "text-base font-medium " +
                              (selectedDate.clone().toDate().getMonth() !==
                              day.getMonth()
                                ? "text-gray-400"
                                : "text-gray-700")
                            }
                          >
                            {day.getDate()}
                          </p>
                        </div>
                      )}
                    </td>
                  ))}
                </tr>
              ))}
            </tbody>
          </table>
        </div>
      </div>
    </div>
  );
}
