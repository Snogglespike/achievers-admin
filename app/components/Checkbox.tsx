interface Props {
  label?: string;
  name: string;
  defaultChecked?: boolean;
  required?: boolean;
  disabled?: boolean;
  readOnly?: boolean;
}

export function Checkbox({ label, name, required, ...props }: Props) {
  return (
    <div className="form-control relative">
      <label
        htmlFor={name}
        className="label cursor-pointer flex-col items-start"
      >
        <span className="label-text mb-1">{label}</span>
        {required && (
          <span className="label-text-alt absolute top-6 left-8 text-2xl text-error">
            *
          </span>
        )}
        <input
          type="checkbox"
          className="checkbox"
          id={name}
          name={name}
          required={required}
          {...props}
        />
      </label>
    </div>
  );
}
