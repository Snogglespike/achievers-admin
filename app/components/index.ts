export * from "./Checkbox";
export * from "./DateInput";
export * from "./Input";
export * from "./Select";

export * from "./Title";
export * from "./SubTitle";
export * from "./Text";

export * from "./LoadingSpinner";
export * from "./Navbar";
export * from "./Drawer";
