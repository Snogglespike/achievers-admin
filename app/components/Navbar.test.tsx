import "@testing-library/jest-dom";

import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";

import { Navbar } from "~/components";

test("Header snapshot", async () => {
  const { baseElement } = render(
    <MemoryRouter>
      <Navbar
        currentUser={{
          appRoleAssignments: [],
          displayName: "",
          givenName: "",
          id: "",
          mail: "",
          surname: "",
          userPrincipalName: "",
          email: "",
        }}
      />
    </MemoryRouter>
  );

  expect(baseElement).toMatchSnapshot();
});
