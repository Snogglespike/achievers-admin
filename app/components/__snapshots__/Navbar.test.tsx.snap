// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar > Navbar snapshot 1`] = `
<body>
  <div>
    <nav
      class="navbar absolute left-0 top-0 h-16 bg-primary text-primary-content shadow-md"
    >
      <div
        class="flex-none lg:hidden"
      >
        <label
          class="btn-ghost btn-square btn"
          for="drawer"
        >
          <svg
            aria-hidden="true"
            class="inline-block h-6 w-6 stroke-current"
            fill="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M3 6.75A.75.75 0 013.75 6h16.5a.75.75 0 010 1.5H3.75A.75.75 0 013 6.75zM3 12a.75.75 0 01.75-.75h16.5a.75.75 0 010 1.5H3.75A.75.75 0 013 12zm0 5.25a.75.75 0 01.75-.75h16.5a.75.75 0 010 1.5H3.75a.75.75 0 01-.75-.75z"
              fill-rule="evenodd"
            />
          </svg>
        </label>
      </div>
      <div
        class="flex-1"
      >
        <a
          class="btn-ghost btn text-xl normal-case"
          href="/"
        >
          <img
            alt="Logo"
            class="mr-4 w-16 rounded"
            src="/images/logo.png"
          />
          <span
            class="text-xl font-semibold tracking-tight"
          >
            Achievers WA
          </span>
        </a>
      </div>
      <div
        class="dropdown-end dropdown hidden lg:block"
      >
        <div
          class="flex cursor-pointer items-center gap-2"
          tabindex="0"
        >
          <div
            class="font-semibold"
          />
          <label
            class="btn-ghost btn-circle avatar btn"
          >
            <div
              class="w-10 rounded-full"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M18.685 19.097A9.723 9.723 0 0021.75 12c0-5.385-4.365-9.75-9.75-9.75S2.25 6.615 2.25 12a9.723 9.723 0 003.065 7.097A9.716 9.716 0 0012 21.75a9.716 9.716 0 006.685-2.653zm-12.54-1.285A7.486 7.486 0 0112 15a7.486 7.486 0 015.855 2.812A8.224 8.224 0 0112 20.25a8.224 8.224 0 01-5.855-2.438zM15.75 9a3.75 3.75 0 11-7.5 0 3.75 3.75 0 017.5 0z"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </label>
        </div>
        <ul
          class="dropdown-content menu rounded-box w-52 bg-base-100 p-2 shadow"
          tabindex="0"
        >
          <li>
            <a
              class="font-semibold"
              href="/logout"
            >
              <svg
                aria-hidden="true"
                class="mr-2 w-6"
                fill="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M7.5 3.75A1.5 1.5 0 006 5.25v13.5a1.5 1.5 0 001.5 1.5h6a1.5 1.5 0 001.5-1.5V15a.75.75 0 011.5 0v3.75a3 3 0 01-3 3h-6a3 3 0 01-3-3V5.25a3 3 0 013-3h6a3 3 0 013 3V9A.75.75 0 0115 9V5.25a1.5 1.5 0 00-1.5-1.5h-6zm5.03 4.72a.75.75 0 010 1.06l-1.72 1.72h10.94a.75.75 0 010 1.5H10.81l1.72 1.72a.75.75 0 11-1.06 1.06l-3-3a.75.75 0 010-1.06l3-3a.75.75 0 011.06 0z"
                  fill-rule="evenodd"
                />
              </svg>
              Logout
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</body>
`;
